pkg/reconcilermanager/controllers/otel_controller_test.go

```diff
@@ -167,7 +521,14 @@ func TestOtelReconcilerGooglecloud(t *testing.T) {
 	gotConfigMap := &corev1.ConfigMap{}
 	err := fakeClient.Get(ctx, cmKey, gotConfigMap)
 	require.NoError(t, err, "ConfigMap[%s] not found", cmKey)
-	asserter.Equal(t, wantConfigMap, gotConfigMap, "ConfigMap")
+
+	// Compare the YAML content as strings
+	gotYAML := gotConfigMapToYAML(gotConfigMap)
+	if configYAML != gotYAML {
```

Comment:
Suggest using `cmp.Diff` to compare strings and print a diff. That will be easier to view the difference compared to the full expected yaml. in other words:

- print the `cmp.Diff` output for easy human readable diff
- print the full actual yaml for easy copy/paste if the developer wants to update the expected yaml

Reply:
Done

Reply:
a few minor nits:
- the typical pattern is `if diff := cmp.Diff(wantYAML, gotYAML); diff != "" { ...`
- use `fmt.Printf`, because `fmt.Println` prints an unintended space between args
- use `t.Fatalf` to fail immediately

Reply:
Done